tests/vehicle_efficiency.cpp

```diff
@@ -0,0 +1,192 @@
+        auto &pt = v->parts[ p ];
+
+       if( pt.is_battery() ) {
+            pt.ammo_set( "battery", pt.ammo_capacity() * veh_fuel_mult );
+        }
+
+        if( pt.is_tank() && v->type->parts[p].fuel != "null" ) {
+            float qty = pt.ammo_capacity() * veh_fuel_mult;
+            qty *= std::max( item::find_type( v->type->parts[p].fuel )->stack_size, 1 );
+            qty /= to_milliliter( units::legacy_volume_factor );
+            pt.ammo_set( v->type->parts[ p ].fuel, qty );
+        }
+    }
+}
+
+const float fuel_level = 0.001;
+
+long test_efficiency( const vproto_id &veh_id, const ter_id &terrain, int reset_velocity_turn, long target_distance )
+{
+    long min_dist = target_distance * 0.99;
+    long max_dist = target_distance * 1.01;
+    clear_game( terrain );
+
+    const tripoint map_starting_point( 60, 60, 0 );
+    vehicle *veh_ptr = g->m.add_vehicle( veh_id, map_starting_point, -90, 0, 0 );
+
+    REQUIRE( veh_ptr != nullptr );
+    if( veh_ptr == nullptr ) {
+        return 0;
+    }
+
+    set_vehicle_fuel( veh_ptr, fuel_level );
+    vehicle &veh = *veh_ptr;
+    const tripoint starting_point = veh.global_pos3();
+    veh.tags.insert( "IN_CONTROL_OVERRIDE" );
+    veh.engine_on = true;
+    REQUIRE( veh.safe_velocity() > 0 );
+    veh.cruise_velocity = veh.safe_velocity();
+    // If we aren't testing repeated cold starts, start the vehicle at cruising velocity.
+    // Otherwise changing the amount of fuel in the tank perturbs the test results.
+    if( reset_velocity_turn == -1 ) {
+        veh.velocity = veh.cruise_velocity;
+    }
+    int reset_counter = 0;
+    long tiles_travelled = 0;
+    int turn_count = 0;
+    while( veh.engine_on && veh.safe_velocity() > 0 ) {
+        g->m.vehmove();
+        veh.idle( true );
+        // How much it moved
+        tiles_travelled += square_dist( starting_point, veh.global_pos3() );
+        // Bring it back to starting point to prevent it from leaving the map
+        const tripoint displacement = starting_point - veh.global_pos3();
+        tripoint veh_pos = veh.global_pos3();
+        g->m.displace_vehicle( veh_pos, displacement );
+        if( reset_velocity_turn < 0 ) {
+            continue;
+        }
+
+        reset_counter++;
+        if( reset_counter > reset_velocity_turn ) {
+            veh.velocity = 0;
+            veh.last_turn = 0;
+            veh.of_turn_carry = 0;
+            reset_counter = 0;
+        }
+    }
+
+    long adjusted_tiles_travelled = tiles_travelled / fuel_level;
+    CHECK( adjusted_tiles_travelled >= min_dist );
+    CHECK( adjusted_tiles_travelled <= max_dist );
+    return adjusted_tiles_travelled;
+}
+
+void find_inner( std::string type, std::string terrain, int delay ) {
+    statistics efficiency;
+    for( int i = 0; i < 10; i++) {
+        efficiency.add( test_efficiency( vproto_id( type ), ter_id( terrain ), delay, 0 ) );
+    }
+    printf( "Testing %s on %s with %s: Min %d, Max %d, Midpoint %f.\n",
```

Comment:
Non-issue: could be replaced with Catch's `INFO( ... )`? That only [prints if next assertion fails](https://github.com/philsquared/Catch/blob/master/docs/logging.md#streaming-macros).

Haven't tried it, not sure that's the intent.